subsys/bluetooth/mesh/adv.h

```diff
@@ -4,6 +4,12 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
+#include <stdint.h>
```

Comment:
Why were these headers added here? I do not see changes in C code.

Reply:
Yeah, sorry for not describing this properly. 

This one is just a simple matter of importing what you use. This file uses `uintX_t` types which are defined by `stdint.h`.

Reply:
Yep, I understand it. I also asked about these below:
```
#include <zephyr/bluetooth/bluetooth.h>
#include <zephyr/sys/util_macro.h>
#include <zephyr/sys_clock.h>
```
why did they become required ones?

Reply:
These header fines types used by the file :) 

A lot of mesh files are generally missing includes which means that if the order of how they are included changes, it randomly breaks things. 

I'm using `clangd` to easily fix missing includes (and recently ensured that all LE Audio files include what they use (https://include-what-you-use.org/)). Mesh is a bit fragile in this regard

Reply:
Yep, sure. I guess it was an initial design idea to include them in correct order in C file to solve such dependencies.
Since initial designers do not maintain source code anymore and current maintainers do not follow the initial idea it seems a bit cumbersome. 
Any case, thanks for clarification.